### PR TITLE
[tf][testnet] enable eks autoscaler

### PR DESCRIPTION
**Pull request by rustielin**
tl;dr this should make Forge cluster setup/cleanup less flaky, increasing visibility of actual test results, and make Forge faster when many tests are being run in succession. Intern testnets will also benefit by having a cheaper "idle" state, something only Forge had in the past.

Enable EKS node autoscaling using Cluster Autoscaler https://docs.aws.amazon.com/eks/latest/userguide/autoscaling.html. Previously, we were handling cluster scaling before and after Forge tests directly within the Forge binary. This was error prone (to EKS connection flakes), and more importantly coupled the results of Forge runs with that of administrative cluster cleanups. This led to some weird concurrency issues that made Forge flaky.

Changes:
* Enable autoscaling using `enable_cluster_autoscaler`, applying the deployment into the `k8s-metrics` chart for logical grouping. This chart is now responsible for all k8s metrics, and metrics-driven autoscaling -- horizontal pod autoscaling (e.g. for CoreDNS), and horizontal node autoscaling (cluster scale).
* Tune the min size for all EKS nodegroups to 1. This should be a no-op for cluster without autoscaling enabled, as EKS will always try to get the desired nodegroup size
* Cluster autoscaler waits 10 minutes before scaling down the cluster, so this should theoretically keep Forge clusters warm for longer, instead of immediately scaling down after a test is done.
* For testnets, all you need to spin it down to an idle state then is to helm uninstall all the relevant aptos releases (validator, testnet), and cluster should scale down! https://github.com/aptos-labs/aptos-core/issues/771

## test

Install autoscaler via the k8s metrics helm chart onto an EKS cluster with 4-validator testnet. Start tailing the logs of the autoscaler while tracking the size of the cluster, and start to uninstall the validators:

```
helm uninstall --keep-history val0
helm uninstall --keep-history val1
helm uninstall --keep-history val2
helm uninstall --keep-history val3
```

By default, cluster autoscaler waits 10 minutes before doing anything. After 10 minutes, the cluster went from 8 nodes (4 validator, 4 utilities) to 3 nodes (1 validator, 2 utilities (to keep the other stuff on my cluster running, like indexer, testnet monitoring))

### CODE REVIEW

**Review by geekflyer — APPROVED** [review]
noice!
